### PR TITLE
BUG: IndexError in libreduction

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @WillAyd @jreback another from the groupby-Exception follow-ups.

### CODE REVIEW

**Comment by WillAyd** [comment]
Any easy way to summarize how this was throwing an IndexError? Not immediately apparent

**Comment by jbrockmendel** [comment]
> Any easy way to summarize how this was throwing an IndexError? Not immediately apparent

Case where `self.bins` is empty, so `self.bins[0]` raises `IndexError`.  The `self.ngroups > 0` check is incorrect because of a kludge on L217-221

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

**Comment by jreback** [comment]
does this affect user facing code in any way?

**Comment by jbrockmendel** [comment]
> does this affect user facing code in any way?

Not AFAICT, no.

**Comment by jreback** [comment]
can you rebase

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -262,10 +262,9 @@ def aggregate(self, func=None, *args, **kwargs):
 
             try:
                 return self._python_agg_general(func, *args, **kwargs)
-            except (AssertionError, TypeError):
+            except (AssertionError, TypeError, IndexError):
```
Comment (jreback) [inline pandas/core/groupby/generic.py]: is there a test change?

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: No, but we currently have tests that raise IndexError that no longer will

**Comment by jbrockmendel** [comment]
Closing for now, I want to take a look at whats happening before we get here

**Pull request closed without merging**